framework-cloud/framework-openshift-operator/src/main/java/org/kie/cloud/openshift/operator/scenario/KieServerWithDatabaseScenarioImpl.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kie.cloud.openshift.operator.scenario;
+
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import cz.xtf.core.waiting.SimpleWaiter;
+import cz.xtf.core.waiting.SupplierWaiter;
+import cz.xtf.core.waiting.WaiterException;
+import org.kie.cloud.api.deployment.ControllerDeployment;
+import org.kie.cloud.api.deployment.DatabaseDeployment;
+import org.kie.cloud.api.deployment.Deployment;
+import org.kie.cloud.api.deployment.KieServerDeployment;
+import org.kie.cloud.api.deployment.SmartRouterDeployment;
+import org.kie.cloud.api.deployment.SsoDeployment;
+import org.kie.cloud.api.deployment.WorkbenchDeployment;
+import org.kie.cloud.api.deployment.constants.DeploymentConstants;
+import org.kie.cloud.api.scenario.KieServerWithDatabaseScenario;
+import org.kie.cloud.openshift.constants.OpenShiftConstants;
+import org.kie.cloud.openshift.deployment.DatabaseDeploymentImpl;
+import org.kie.cloud.openshift.deployment.KieServerDeploymentImpl;
+import org.kie.cloud.openshift.operator.deployment.KieServerOperatorDeployment;
+import org.kie.cloud.openshift.operator.model.KieApp;
+import org.kie.cloud.openshift.operator.model.components.Auth;
+import org.kie.cloud.openshift.operator.model.components.Server;
+import org.kie.cloud.openshift.operator.model.components.Sso;
+import org.kie.cloud.openshift.util.SsoDeployer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class KieServerWithDatabaseScenarioImpl extends OpenShiftOperatorScenario<KieServerWithDatabaseScenario> implements KieServerWithDatabaseScenario {
+
+    private KieServerDeploymentImpl kieServerDeployment;
+    private DatabaseDeploymentImpl databaseDeployment;
+    private boolean deploySso;
+    private SsoDeployment ssoDeployment;
+
+    private static final Logger logger = LoggerFactory.getLogger(KieServerWithDatabaseScenarioImpl.class);
+
+    public KieServerWithDatabaseScenarioImpl(KieApp kieApp, boolean deploySso) {
```

Comment:
Just for case we do not forgot, there will be probably need to be done rebase of this PR because of changes done in #591 
Instead of boolean there is used object with more possible configuration ScenarioRequest

Reply:
This change is done in https://github.com/kiegroup/kie-cloud-tests/pull/591. So, if this PR is merged first, we don't need to fix the conflicts, otherwise I'm happy to fix them, no problemo.